sycl-fusion/jit-compiler/lib/KernelFusion.cpp

```diff
@@ -191,3 +188,14 @@ FusionResult KernelFusion::fuseKernels(
 
   return FusionResult{FusedKernelInfo};
 }
+
+void jit_compiler::resetConfiguration() { ConfigHelper::reset(); }
+
+template <typename Opt>
+void jit_compiler::set(const typename Opt::ValueType &Value) {
+  ConfigHelper::getConfig().set<Opt>(Value);
+}
+
+template void jit_compiler::set<option::JITEnableVerbose>(const bool &Value);
+template void jit_compiler::set<option::JITEnableCaching>(const bool &Value);
+template void jit_compiler::set<option::JITTargetInfo>(const TargetInfo &Value);
```

Comment:
Ideally, it would be enough to define everything related to a certain option in `Options.h`, so these explicit instantiations here is a bit sub-optimal... WDYT, is there a better way?

Reply:
I don't think we are gaining much from making all these templates. We could drop `OptionBase` and use [LLVM RTTI](https://llvm.org/docs/HowToSetUpLLVMStyleRTTI.html) to declare a:

```c++
void jit_compiler::set(OptionPtrBase &&Option);
```

In the end, we are just storing the pointer base to the map, right? We could keep the current interface, but add to the headers:
```c++
template <typename Opt, typename... Args>
void jit_compiler::set(Args&&... As) {
  set(Opt::get(std::forward<Args>(As)...);
}
```
Calling the above method (private).

WDYT?

Reply:
> In the end, we are just storing the pointer base to the map, right?

I think this is the key insight. I was unsure about adding LLVM-based RTTI because `Options.h` is included by `jit_compiler.cpp`, but with `OptionID` we already have something to identify the different option classes. I'm now just storing it in `OptionPtrBase`, to be able to query it when entering an option into the map. As a result, defining a new options happens solely in `Options.h`.

Reply:
Spot on. That's what I meant :+1: